lib/s3routes/routes.js

```diff
@@ -35,12 +35,11 @@ function checkUnsupportedRoutes(reqMethod) {
 
 function checkBucketAndKey(bucketName, objectKey, method, reqQuery,
     blacklistedPrefixes, log) {
-    // if empty name and request not a list Buckets
-    if (!bucketName && !(method === 'GET' && !objectKey)) {
+    const getServiceCall = (method === 'GET' && !bucketName && !objectKey);
```

Comment:
you may remove the superfluous `&& !bucketName` test (as far as I understand the naming would be still ok since a GET with a bucket name is also a service call).

Reply:
`getService` is the name AWS uses for the list buckets call, when you make a GET request without defining a bucketname or object key resource in the request http://docs.aws.amazon.com/AmazonS3/latest/API/RESTServiceGET.html
but I know I can remove the test since the check below also checks for bucketName, I was just upset about naming the variable `getServiceCall` without including the bucketName check in there. :(